swim/memberlist.go

```diff
@@ -279,6 +281,83 @@ func (m *memberlist) SetLocalStatus(status string) {
 	m.postLocalUpdate()
 }
 
+func (m *memberlist) SetLocalLabel(key, value string) error {
+	// TODO implement a sane limit for the size of the labels to prevent users
+	// from impacting the performance of the gossip protocol.
+
+	// ensure that there is a labels map
+	if m.local.Labels == nil {
```

Comment:
I think locking makes sense. The SetLocalState should share the same lock, and potentially the bumIncarnation as well.